lessons-planer/lessons-planer
# #10 submit replacement form

##CHANGES:
1. Implemented logic for refresh token. So files to be checked: 
    app\src\redux\actions\auth.js
    especially refresh_token function

    app\src\redux\reducers\auth.js
    especially TOKEN_REFRESH_SUCCESS and TOKEN_REFRESH_FAIL

    app\src\redux\actions\types.js
    is the naming right or would you change it for some reason?

2. Code clean-up

##HOW TO TEST?
1. checkout to master
2. git pull
3. npm i
4. checkout to '#10_submitReplacementForm'
5. npm start
6. login using mock credentials
7. wait for 5 minutes (access token will expire after that).
8. Have you been redirected to login? If no, then it works :D
9. open redux dev tools to check if actions order is correct
10. open application tab in dev tools to check if cookies are changing once TOKEN_REFRESH_SUCCESS action is done.

##QUESTIONS:
1. Is the code safe?
2. Did you find anything vulnerable?
3. Is the logic implemented correctly or should it be changed?
